### ISSUE TITLE
[Android] Impossible to show modal scrim

### ISSUE THREAD

**Issue by evancharlton** [facebook/react-native#39018]
### Description

The scrim on the `Modal` component is impossible to use on Android.

Here's a screenshot of the UI we're attempting to recreate:
![image](https://github.com/facebook/react-native/assets/418560/acf7fc56-62e3-4920-b21c-c76f9ab07890)

Just a simple dialog, with a scrim partially obscuring the content behind it.

In theory, should be possible with something like:
```ts
<Modal /* snip */>
  <Text>This is some text</Text>
  <Button title="Close" /* snip */ />
</Modal>
```

However, this doesn't work. We're given a white backdrop that covers the whole screen.
![image](https://github.com/facebook/react-native/assets/418560/578b8417-2fbf-40ac-9ec0-ca71bb6a2c37)

If we pass `transparent={true}` to remove the background, the dialog's backdrop is 100% transparent, showing the content behind it.
![image](https://github.com/facebook/react-native/assets/418560/c6c281a2-632f-461f-8a85-e607ca07932c)

Alright, so we need to draw our own scrim - that's simple enough.
![image](https://github.com/facebook/react-native/assets/418560/0ffea7b9-254e-4e65-9f3e-d03a0c27fe5a)

However, this results in the scrim not covering the status (notification) bar and the gesture indicator at the bottom. Both of these parts of the screen outside of a typical app's window, and can't be drawn on.

----

There are a few issues in the underlying code, which I'll enumerate here.

## Forcing white background

The white background is forced by the Modal, on [this line of code][white-bg]. It's possible to remove this white background by passing `transparent={true}`, but that uncovers another bug.

## Removing the scrim on transparent dialogs

When the dialog is rendered with `transparent={true}`, then the modal is rendered without a background. However, when this happens, the `FLAG_DIM_BEHIND` is cleared from the Dialog, which removes the scrim. This happens on [this line of code][dim-behind].

Investigating _that_ uncovers another bug.

## The dialog is never updated

When investigating that bug, I discovered that `updateProperties()` is _always_ short-circuited. Specifically, the `window.isActive()` check on [this line of code][window-active] is always `false`.

Removing this fixed the issue. However, it was added in https://github.com/facebook/react-native/commit/7abcaafd6600535825aa8330af7290ba8acea245, which alluded to a crashing issue. I'm not sure whether this _specific_ check was out of paranoia, or what, but it seemed risky to undo without knowing more.

[dim-behind]:
  https://github.com/facebook/react-native/blob/540c41be91ea6bf318190098dfc88ebdecc273a8/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.java#L389

[window-active]:
  https://github.com/facebook/react-native/blob/540c41be91ea6bf318190098dfc88ebdecc273a8/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.java#L376

[white-bg]:
  https://github.com/facebook/react-native/blob/540c41be91ea6bf318190098dfc88ebdecc273a8/packages/react-native/Libraries/Modal/Modal.js#L228

### React Native Version

main (0.72+)

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
  OS: macOS 13.5
  CPU: (10) arm64 Apple M1 Max
  Memory: 202.89 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.1
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 9.6.7
    path: /usr/local/bin/npm
  Watchman:
    version: 2023.08.07.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK:
    Android NDK: 25.2.9519653
IDEs:
  Android Studio: 2022.1 AI-221.6008.13.2211.9619390
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.8
    path: /opt/homebrew/opt/openjdk@17/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

### Steps to reproduce

Attempt to create the above dialog. I believe that [the Snack][snack] and the steps above are sufficient for this.

[snack]: https://snack.expo.dev/@echarlton/modal

### Snack, screenshot, or link to a repository

Snack to demonstrate this: https://snack.expo.dev/@echarlton/modal

**Comment by evancharlton** [comment]
Version info added. This is reproducible on `main`, too.

**Comment by ravirajn22** [comment]
Yes, Modal is little confusing. But temporarily you can use `statusBarTranslucent` property.

Note: If you use this property then in the Modal android:windowSoftInputMode="adjustResize will not work. You have to handle keyboard space.

**Comment by hanwenbo** [comment]
statusBarTranslucent can help u ?

**Comment by AlixH** [comment]
Scrim is broken on RN 0.83 Dialog
